### PR TITLE
[ONNX] Expand `_cast_` symbolic functions

### PR DESCRIPTION
**Pull request by justinchuby**
The `_cast_` family of symbolic functions has been created from a template function. Even though it saved some lines, it very much obscured the intention of the code. Since the list doesn't really change and the `_cast_` family are IIRC deprecated, it is safe for us to expand the templates and make the code more readable.

This PR also removes any direct calls to `_cast_` functions to maintain a consistent pattern of directly creating `Cast` nodes.

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -3376,51 +3376,58 @@ def _unique2(g: jit_utils.GraphContext, input, sorted, return_inverse, return_co
...
-# def _cast_Byte
-# def _cast_Char
-# def _cast_Short
-# def _cast_Int
-# def _cast_Long
-# def _cast_Half
-# def _cast_Float
-# def _cast_Double
-# def _cast_ComplexFloat
-# def _cast_ComplexDouble
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset9.py:3398]: Removed because they do not exist

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -3376,51 +3376,58 @@ def _unique2(g: jit_utils.GraphContext, input, sorted, return_inverse, return_co
...
-    func_name = f"_cast_{scalar_type}"
-    globals()[func_name] = _onnx_symbolic(f"aten::{func_name}")(
-        symbolic_helper.parse_args("v", "i")(
-            functools.partial(
-                _cast_func_template,
-                _type_utils.JitScalarType.from_name(scalar_type).onnx_type(),
-            )
-        )
-    )
+def _cast_Byte(g: jit_utils.GraphContext, input, non_blocking):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:3381]: Should we add deprecation warning? (although the only reason they need to exist is for any potential bc code that no one looks at) wdyt?

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -4752,7 +4759,7 @@ def _pack_padded_sequence(g: jit_utils.GraphContext, input, lengths, batch_first
     # It's really only necessary because those operators expand to something that
     # only works with int32 types in Caffe2...
     if lengths.type().scalarType() != "Int":
-        lengths = _cast_Int(g, lengths, False)  # type: ignore[name-defined]
+        lengths = g.op("Cast", lengths, to_i=_C_onnx.TensorProtoDataType.INT32)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:4762]: `_C_onnx.TensorProtoDataType.INT32`

I'm starting to think if we should have wrapper classes to hide away the `_C.onnx`, and likewise `_C.torch` details.

**Review by BowenBao — APPROVED** [review]
LGTM, some related thoughts in comment, not blocking.

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:3381]: We should but that would also cause the exporter to vomit if a model is too old. Is that something we care about?

File: torch/onnx/symbolic_opset9.py (hunk 3 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:4762]: Maybe replace everything with JitScalarType then?

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:3381]: I added deprecatoin message. Let's see how it goes

**Pull request closed without merging**